spinalcordtoolbox/registration/core.py

```diff
@@ -339,6 +343,8 @@ def register(src, dest, step, param):
 
     # # landmark-based registration
     if step.type in ['label']:
+        if step.algo:
```

Comment:
👍

Reply:
Ah, one thing to note here: 

Even if `algo` isn't explicitly specified in the `-param` command, `sct_register_multimodal` will default to `algo=syn`, which meaning that the warning will always trigger. For example:

```
sct_register_multimodal -param step=0,type=label
```

![image](https://user-images.githubusercontent.com/16181459/193658884-9fe12013-ac8f-4a9a-a76c-7eb3522a722f.png)

Maybe the warning is still worth including?

But, at the same time, I think it speaks to what I wrote in https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3893#issuecomment-1265512260, too, in that it's a bit weird that `type='label'` acts like an `algo`, but isn't.

Reply:
if we decide to attack this beast, we should also include the other (related) issues, eg: when selecting algo=syn, the field 'poly' has nothing to do with it. There are many other examples like that...

Reply:
From your screencap, it looks like the warning is coloured, but then I don't understand your commit message:

> Is printv the right function for this? We're in the API, so maybe
logging should be used instead. But, I like the colors...

Is the screencap from a previous version of the code?

Regardless, I think it's fine to use `printv` in this case, since the rest of the function uses `printv` a lot. Consistency is very important! It matters most within a single function, and a lot within a single file. If/when this part of the code gets an overhaul, we can decide how to deal with the fact that it's calling `printv` at the API layer. Possibly the chatty parts of the code would move out of the API; or possibly all of it would get converted to logging instead.

(Sorry for the self-contradiction with my previous statements about `printv`! It's just that _either_ `printv` _or_ logging is better than a mix of both within the same function/file.)

Reply:
Ah gosh, sorry. My commit message has made a mess of things. Let me try and clear up what happened:

* I originally used `printv`, thinking it was the only way to get yellow text (hence the commit message)
* I then realized that Pytest's `caplog.text` only captures logging warnings/messages, so I swapped out `printv` for `logging`, then amended the commit (while forgetting to amend the commit message!)
* I _then_ realized that `logger.warning` actually _does_ print yellow text too (which is what you see in the screencap). The fact that they both print yellow text makes the commit message seem even more misleading.

I'm happy to switch to using `printv` in the API function, I'd just need to find a way for Pytest to pick up that sort of output. :)